dscache/dscache.go

```diff
@@ -18,28 +22,48 @@ var (
 
 // Dscache represents an in-memory serialized dscache flatbuffer
 type Dscache struct {
+	Filename            string
 	Root                *dscachefb.Dscache
 	Buffer              []byte
 	ProfileIDToUsername map[string]string
 }
 
-// LoadDscacheFromFile will load a dscache from the given filename
-func LoadDscacheFromFile(filename string) (*Dscache, error) {
-	buffer, err := ioutil.ReadFile(filename)
+// NewDscache will construct a dscache from the given file, or will construct an empty dscache
+// that will save to the given filename
+func NewDscache(fsys qfs.Filesystem, filename string) *Dscache {
+	ctx := context.TODO()
+	f, err := fsys.Get(ctx, filename)
 	if err != nil {
-		return nil, err
+		// Ignore error, as dscache loading is optional
+		return &Dscache{Filename: filename}
+	}
+	defer f.Close()
+	buffer, err := ioutil.ReadAll(f)
+	if err != nil {
+		// Ignore error, as dscache loading is optional
+		return &Dscache{Filename: filename}
 	}
 	root := dscachefb.GetRootAsDscache(buffer, 0)
-	return &Dscache{Root: root, Buffer: buffer}, nil
+	return &Dscache{Filename: filename, Root: root, Buffer: buffer}
+}
+
+// IsEmpty returns whether the dscache has any constructed data in it
+func (d *Dscache) IsEmpty() bool {
```

Comment:
can we make `Dscache` nil-callable? We could do this in a follow-up PR, but given that we're modifying the repo interface to include one, I'd love to be able to wire up a nil cache & not have everything explode. This test spells it out:

https://github.com/qri-io/qri/blob/e3086cfd1944cc782f960356cb4bf6bdca618bfb/logbook/logbook_test.go#L183-L232

Many pointer methods are calling `IsEmpty` as a guard, which is already halfway there. Could also add:
```go
func (d *Dscache) IsEmpty() bool {
  if d == nil {
    return true
  }
  return d.Root == nil
}
```

for all non-error-return funcs

Reply:
Easy enough, done.